### ISSUE TITLE
Upgrade to MODX 3: Text TVs cause "HTTP/2 500 Internal Server Error"

### ISSUE THREAD

**Issue by jolichter** [modxcms/revolution#16329]
### Summary
I upgraded MODX 2.8.4-pl via UpgradeMODX to 3.0.0-pl and then to 3.0.2-pl (core/cache cleaned).
When I try to edit a resource which use Text TVs, I get a blank screen. The browser console reveals:

```
GET https://myDomain.tld/manager/?a=resource/update&id=349
Status: 500 Internal Server Error
Version: HTTP/2
Transferred: 389 B (0 B size)
Referrer Policy: strict-origin-when-cross-origin
Request Priority: Highest
```

### Step to reproduce
1. Create a TV with input type Text
2. Add TV to a template
3. Edit a resource in menu tree (which use this template)
4. MODX Manager crashes (blank page)

This is caused only by simple text TVs (Text input types). No Error Log. Editing a resource via Qiuck Edit works (I guess because it doesn't use TVs).

Addition: If I change the input types e.g. from Text to RichText, no error appears.

This had been working with MODX 2.x.x versions without any problems and I do not have this problem with a fresh new 3.0.2-pl installation. Therefore I think that something went wrong during the upgrade. However, there was no error message and I had tried it 2 times with my MODX backup.

### Questions
Is anything like this reported? Does anyone know how to fix this?

### Environment
- MODX 3.0.2-pl
- PHP 7.4.16
- MySQL 5.7.33
- TinyMCE RTE 2.0.9-pl
- FormIt 5.0.1-pl
- pdoTools 3.0.1-pl
- pThumb 2.3.3-pl
- Resizer 1.0.2-beta
- UpgradeMODX 2.3.3-pl

**Comment by Jako** [comment]
Please look into the PHP error log of that installation. There you can see the cause of the error 500. The location can be found in the phpinfo() page (a clickable link exists in manager/?a=system/info) in the line `error_log`.

**Comment by jolichter** [comment]
Hi Jako, I found this two errors in the `error_log`:

```
[authz_core:error] [pid 25969] [client xxx.xxx.xxx.xxx:55570] AH01630: client denied by server configuration: /var/www/vhosts/myServer.tld/httpdocs/modx/core/docs/changelog.txt

[proxy_fcgi:error] [pid 25994] [client xxx.xxx.xxx.xxx:56810] AH01071: Got error 'PHP message: PHP Fatal error: Cannot declare class modTemplateVarInputRenderText, because the name is already in use in /var/www/vhosts/web.myServer.tld/httpdocs/modx/core/src/Revolution/Processors/Element/TemplateVar/Renders/mgr/input/text.class.php on line 17', referer: https://myDomain.tld/manager/?a=element/tv/update&id=42
```

Think the first error message has nothing to do with it? See this also in other MODX installations. 
Is the second helpful? The name is already in use? ID42 is in that case my Text TV named "Corner". I use only one TV which is called "Corner" and a rename doesn't fix it.

**Comment by jolichter** [comment]
Weird, I found this issue on [modx.pro/help/4986](https://modx.pro/help/4986) and have modified my `text.class.php`:

```
<?php
/*
 * This file is part of MODX Revolution.
 *
 * Copyright (c) MODX, LLC. All Rights Reserved.
 *
 * For complete copyright and license information, see the COPYRIGHT and LICENSE
 * files found in the top-level directory of this distribution.
 */

use MODX\Revolution\modTemplateVarInputRender;

/**
 * @package modx
 * @subpackage processors.element.tv.renders.mgr.input
 */

# Prevents error "Got error PHP message: PHP Fatal error: Cannot declare class modTemplateVarInputRenderText, because the name is already in use", add exist the class already?
if(!class_exists('modTemplateVarInputRenderText')){
    class modTemplateVarInputRenderText extends modTemplateVarInputRender {
        public function getTemplate() {
            return 'element/tv/renders/input/textbox.tpl';
        }
    }
}
return 'modTemplateVarInputRenderText';
```

Now it works. However, the cause is unclear to me and why this is only the case in my MODX upgrade.
A mistake on my part or a bug?

**Comment by jolichter** [comment]
OK, I think I got it and it is an upgrade bug. I checked my DB via SQL-Dump and found this for instance (sql sample):

```

INSERT INTO `modx_media_sources_elements` (`source`, `object_class`, `object`, `context_key`) VALUES
......

(1, 'modTemplateVar', 40, 'web'),
(1, 'modTemplateVar', 40, 'web2'),
(1, 'modTemplateVar', 40, 'web4'),
(1, 'modTemplateVar', 41, 'web'),
(1, 'modTemplateVar', 41, 'web2'),
(1, 'modTemplateVar', 41, 'web4'),
(1, 'modTemplateVar', 42, 'web'),
(1, 'modTemplateVar', 42, 'web2'),
(1, 'modTemplateVar', 42, 'web4'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web2'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web4'),

......
```
Fix: Change all `modTemplateVar` to `MODX\\Revolution\\modTemplateVar` and delete duplicate modTemplateVar, in this case my TV ID 42. Should look like this:

```
INSERT INTO `modx_media_sources_elements` (`source`, `object_class`, `object`, `context_key`) VALUES
......

(1, 'MODX\\Revolution\\modTemplateVar', 40, 'web'),
(1, 'MODX\\Revolution\\modTemplateVar', 40, 'web2'),
(1, 'MODX\\Revolution\\modTemplateVar', 40, 'web4'),
(1, 'MODX\\Revolution\\modTemplateVar', 41, 'web'),
(1, 'MODX\\Revolution\\modTemplateVar', 41, 'web2'),
(1, 'MODX\\Revolution\\modTemplateVar', 41, 'web4'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web2'),
(1, 'MODX\\Revolution\\modTemplateVar', 42, 'web4'),

......
```

Now it works as expected without modifying `text.class.php`. Maybe anyone else can also confirm this upgrade issue? If so, can someone please fix this?

PS: If someone doesn't find an error in his DB, modifying the `text.class.php` should help as a work around (until the next update ;-)

**Comment by jolichter** [comment]
Strange, I've updated a second MODX 2.8.4 installation to MODX 3.0.3 and have the same problem as described above, except that I am using PHP 8.1.16, MariaDB 10.3.38 with MODX 3.0.3-pl. The SQL database looks OK, 'modTemplateVar' has been replaced by 'MODX\\Revolution\modTemplateVar'. 

PHP Error Log:
`Got error 'PHP message: PHP Fatal error: Cannot declare class modTemplateVarInputRenderText, because the name is already in use in /var/www/vhosts/web.myServer.tld/httpdocs/modx/core/src/Revolution/Processors/Element/TemplateVar/Renders/mgr/input/text.class.php on line 0', referer: https://myDomain.tld/manager/?a=element/tv/update&id=78`

Therefore I have modified the `text.class.php` as described above.

This is a workaround but does not fix the reason? How can I find the reason for this, something about my database is not correct?

With this workaround my MODX 3 is running and I don't see any errors in the MODX Error Log.
